### PR TITLE
Add docker container for CentOS8

### PR DESCRIPTION
**Pull request by markrtuttle**
This pull request adds a docker container for CentOS8 to the docker branch of the template repository.

Specifically, the three commits
* Move preexisting docker work done for the recent hackathon into its own directory
* Add a CentOS8 container with installed CBMC and creating a cbmc-centos8.tar.gz tarball
* Add an Ubuntu18 container with installed CBMC and creating a cbmc-ubuntu18.tar.gz tarball

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+This directory contains docker files for building cbmc inside a docker
+container for various operating systems.
+
+* The Dockerfile builds a container with
+    * A CBMC installation: CBMC is intalled in the operating system's
+      equivalent of /usr/local/bin
+    * A CBMC tarball: CBMC binaries are packaged into a tarball in the
+      operating system's equivalent of /tmp/cbmc.tar.gz.
+
+* The Makefile contains two targs:
```
Comment (danielsn) [inline docker/README.md:10]: targs?

File: docker/ubuntu18/Makefile [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+USER = `whoami`
+GROUP = `groups | awk '{print $$1}'`
+
+default: container
+
+container:
+	docker build --tag cbmc-$(OS) .
+
+tarball:
+	docker run -v `pwd`:/home cbmc-$(OS) cp /tmp/$(TARBALL) /home
```
Comment (danielsn) [inline docker/ubuntu18/Makefile:13]: Should this depend on `container?`

File: docker/README.md (hunk 1 above)
Comment (markrtuttle) [reply docker/README.md]: Changed to "targets"

File: docker/ubuntu18/Makefile (hunk 2 above)
Comment (markrtuttle) [reply docker/ubuntu18/Makefile:13]: I'm reluctant to have one phony target depend on another.  If the container target built something that could be tested, I'd be more inclined.

**Pull request merged**